std/range/package.d

```diff
@@ -7125,9 +7131,64 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Impl* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
+        @property auto front() { return impl.r.front; }
+        void popFront()
+        {
+            assert(impl.curSizeLeft > 0 && !impl.r.empty);
+            impl.curSizeLeft--;
+            impl.r.popFront();
+        }
+    }
+
+    static struct Impl
+    {
+        private Source r;
+        private size_t chunkSize;
+        private size_t curSizeLeft;
+        private bool _empty;
+    }
+
+    private Impl* impl;
+    alias impl this;
+
+    private this(Source r, size_t chunkSize)
+    {
+        impl = new Impl(r, chunkSize, chunkSize);
```

Comment:
I'm on the fence about this.  Ideally, we should use `RefCounted` here, I think `byChunk` does that, but then it's a toss-up between whether you want to keep `@nogc` or `@safe`, but you can't have both.

I'm OK either way -- if you guys feel it's more consistent to use `RefCounted` ala `byChunk`, then it's a relatively easy change.